discojs/discojs-core/src/dataset/preprocessing/text_preprocessing.ts

```diff
@@ -0,0 +1,30 @@
+import { tf, Task } from '../..'
+import GPT3Tokenizer from 'gpt3-tokenizer'
+
+type TextEntry = string | string[]
+type PreprocessText = (text: TextEntry) => tf.TensorContainer
+
+export interface TextTensorContainer extends tf.TensorContainerObject {
+  xs: tf.Tensor1D
+  ys: tf.Tensor1D | number | undefined
+}
+
+export enum TextPreprocessing {
+  Tokenize = 'tokenize'
+}
+
+export function getPreprocessImage (task: Task): PreprocessText {
```

Comment:
should this one be called image?

Reply:
also mind adding a comment if you will output a stream of token ids?

for LLMs, we can then also support datasets without any label being needed

also let's say where/how people could load different tokenizers (task config or hardcoded either is fine)

Reply:
I'll make sure that the PR follows your comments once it's out of the "draft" stage!